sdk/storage/src/common/reliable_stream.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "common/reliable_stream.hpp"
+#include "http/http.hpp"
+
+using Azure::Core::Context;
+using Azure::Core::Http::BodyStream;
+
+namespace Azure { namespace Storage {
+
+  int64_t ReliableStream::Read(Context& context, uint8_t* buffer, int64_t count)
+  {
+    if (this->m_options.DoInjectError)
+    {
+      throw std::runtime_error("Injected error");
+    }
+
+    for (int64_t intent = 1;; intent++)
+    {
+      // check if we need to get inner stream
+      if (this->m_inner == nullptr)
+      {
+        // Get a a bodyStream that starts from last known offset
+        // if this fails, throw bubbles up
+        // As m_inner is unique_pr, it will be destructed on reassignment, cleaning up network
+        // session.
+        this->m_inner = this->m_httpGetter(context, this->m_retryInfo);
+      }
+      try
+      {
+        auto const readBytes = this->m_inner->Read(context, buffer, count);
+        // update offset
+        this->m_retryInfo.Offset += readBytes;
+        return readBytes;
+      }
+      catch (std::runtime_error const& e)
+      {
+        // forget about the inner stream. We will need to request a new one
+        // As m_inner is unique_pr, it will be destructed on reassignment (cleaning up network
+        // session).
+        this->m_inner.release();
+        (void)e; // todo: maybe log the exception in the future?
```

Comment:
Nit: i remember compiler will never give a warning for unused exception.

Reply:
well, that makes me wonder about Storage code's warnings level...  `/W4` should be applying every where, and that one will complain about this like this: https://dev.azure.com/azure-sdk/public/_build/results?buildId=480284&view=logs&jobId=cc13375d-86b2-59e0-d013-a4f72ffc9bbe&j=cc13375d-86b2-59e0-d013-a4f72ffc9bbe&t=3b69ab8d-74c6-505f-b364-719e5a272167

Reply:
looks VC++ treats these exception variables differently from GCC does.